dockstore-client/src/main/java/io/dockstore/client/cli/nested/AbstractEntryClient.java

```diff
@@ -770,6 +784,38 @@ protected void testParameter(List<String> args) {
         return Optional.empty();
     }
 
+    /**
+     * Returns the correct path for use in TRS endpoints
+     *
+     * @param entryPath path to either a tool or workflow
+     */
+    public String getTrsId(String entryPath) {
+        return entryPath;
+    }
+
+    /**
+     * Get all tool descriptors associated with the entry type
+     * @param type descriptor type, CWL, WDL, NFL ...
+     * @param entryPath path to either a tool or workflow, the path for a workflow must have the #workflow/ prefix
+     * @param versionID version we are fetching descriptors for
+     */
+    public List<ToolFile> getAllToolDescriptors(String type, String entryPath, String versionID) {
+        final Ga4Ghv20Api ga4ghv20api = this.getClient().getGa4Ghv20Api();
+
+        // get all the tool files and filter out anything not a descriptor
+        try {
+            return ga4ghv20api.toolsIdVersionsVersionIdTypeFilesGet(type, entryPath, versionID).stream()
+                .filter(toolFile -> ToolFile.FileTypeEnum.SECONDARY_DESCRIPTOR.equals(toolFile.getFileType()) || ToolFile.FileTypeEnum.PRIMARY_DESCRIPTOR.equals(toolFile.getFileType()))
+                .collect(Collectors.toList());
+        } catch (NullPointerException ex) {
```

Comment:
Have you seen an NPE happen? You need to catch if so, but just wondering why you need this.

Reply:
This was, when looking back, caused by an issue addressed in an earlier comment. By changing the order in which `ToolFile.FileTypeEnum.SECONDARY_DESCRIPTOR.equals(toolFile.getFileType())` is checked, the NPE is no longer present. Removed the catch.